OpenRA.Mods.Common/Scripting/Global/PaletteGlobal.cs

```diff
@@ -0,0 +1,33 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Scripting;
+
+namespace OpenRA.Mods.Common.Scripting
+{
+	[ScriptGlobal("Palette")]
+	public class PaletteGlobal : ScriptGlobal
+	{
+		FlashPaletteEffect fpe;
+
+		public PaletteGlobal(ScriptContext context)
+			: base(context)
+		{
+			fpe = context.World.WorldActor.TraitOrDefault<FlashPaletteEffect>();
```

Comment:
You could add a type argument to FlashEffect / Type field to FlashPaletteEffectInfo, cache an array of all traits, and then filter for the requested trait in FlashEffect.